osquery/core/conversions.h

```diff
@@ -322,4 +306,102 @@ inline typename std::enable_if<
 tryTo(FromType&& from) {
   return std::forward<FromType>(from);
 }
+
+/**
+ * tryTo [w]string to integer implementation
+ * int
+ * long int
+ * long long int
+ * unsigned int
+ * unsigned long int
+ * unsigned long long int
+ * Do not use throwingStringToIntImpl, it throws exception in case of error
+ */
+template <typename ToType, typename FromType>
+inline
+    typename std::enable_if<std::is_same<ToType, int>::value &&
+                                (std::is_same<FromType, std::string>::value ||
+                                 std::is_same<FromType, std::wstring>::value),
+                            ToType>::type
```

Comment:
Usually readers find these kinds of SFINAE trickery hard to understand and too verbose - you can hide most uninteresting things inside a [helper type-trait](https://godbolt.org/g/7UQyei):

```cpp
template <typename FromType, typename CallerType, typename AllowedType>
struct ConversionAllowed
{
	static constexpr bool value = std::is_same<CallerType, AllowedType>::value &&
		(std::is_same<FromType, std::string>::value ||
			std::is_same<FromType, std::wstring>::value);
};
```
You can reuse it in almost all of `throwingStringToIntImpl` overloads like this:

```cpp
template <typename ToType, typename FromType>
inline typename std::enable_if<ConversionAllowed<FromType, ToType, int>::value, ToType>::type
throwingStringToIntImpl(const FromType& from, const int base)
{
}
```

Reply:
Also you probably wouldn't want to expose that helper to outside callers - usually people use a `impl` or `details` namespace to indicate that something is not for outside use. `throwingStringToIntImpl` can also go into that namespace and this way you can probably omit comments saying "do not use this"

Reply:
Good idea with creating additional type-trait! Will do it, but a little bit different way: I am going to create a `IsStlString` and `IsInteger`. Which can make this code clear and also permit us use this traits somewhere else if need.

Also, will pull all implementation code in `impl` namespace. Thanks for it!